From 7f3bc63ca42b9dbe370940643a492456f6c91bab Mon Sep 17 00:00:00 2001
From: Michael Sweet <michael.r.sweet@gmail.com>
Date: Mon, 11 Sep 2017 17:08:17 -0400
Subject: =?UTF-8?q?The=20scheduler=20(incorrectly)=20woke=20up=20once=20pe?=
 =?UTF-8?q?r=20second=20to=20remove=20stale=20temporary=0Aqueues=20(Issue?=
 =?UTF-8?q?=20#5100).?=

- scheduler/main.c: Update local_timeout to start at 0 and only get updated as
  needed.

Fixes: #5100
---
 scheduler/main.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/scheduler/main.c b/scheduler/main.c
index 3a51e7bea..3e899e347 100644
--- a/scheduler/main.c
+++ b/scheduler/main.c
@@ -710,7 +710,7 @@ main(int  argc,				/* I - Number of command-line args */
   current_time  = time(NULL);
   event_time    = current_time;
   expire_time   = current_time;
-  local_timeout = current_time + 60;
+  local_timeout = 0;
   fds           = 1;
   report_time   = 0;
   senddoc_time  = current_time;
@@ -972,7 +972,10 @@ main(int  argc,				/* I - Number of command-line args */
     */
 
     if (current_time >= local_timeout)
+    {
       cupsdDeleteTemporaryPrinters(0);
+      local_timeout = 0;
+    }
 
 #ifndef HAVE_AUTHORIZATION_H
    /*
@@ -1780,11 +1783,11 @@ select_timeout(int fds)			/* I - Number of descriptors returned */
 
   for (printer = (cupsd_printer_t *)cupsArrayFirst(Printers); printer; printer = (cupsd_printer_t *)cupsArrayNext(Printers))
   {
-    if (printer->temporary && !printer->job && local_timeout > (printer->state_time + 60))
+    if (printer->temporary && !printer->job && (!local_timeout || local_timeout > (printer->state_time + 60)))
       local_timeout = printer->state_time + 60;
   }
 
-  if (timeout > local_timeout)
+  if (timeout > local_timeout && local_timeout)
   {
     timeout = local_timeout;
     why     = "delete stale local printers";
